test/boost/reader_concurrency_semaphore_test.cc

```diff
@@ -844,6 +844,7 @@ SEASTAR_THREAD_TEST_CASE(test_reader_concurrency_semaphore_admission) {
         auto permit2 = semaphore.obtain_permit(schema, get_name(), 1024, db::timeout_clock::now(), {}).get();
         auto irh2 = semaphore.register_inactive_read(make_empty_flat_reader_v2(s.schema(), permit2));
 
+        sleep(50ms).get();
```

Comment:
Looks like a flaky test was born here.

Reply:
Maybe you can add a test-only method to the semaphore, which would allow hooking into the loop wake-ups.

Reply:
I realised, I can replace it with `REQUIRE_EVENTUALLY_EQUAL(bool(irh1), false);`